tq/basic_upload.go

```diff
@@ -215,3 +215,21 @@ func configureBasicUploadAdapter(m *Manifest) {
 		return nil
 	})
 }
+
+func (a *basicUploadAdapter) makeRequest(t *Transfer, req *http.Request) (*http.Response, error) {
+	res, err := a.doHTTP(t, req)
+	if errors.IsAuthError(err) && len(req.Header.Get("Authorization")) == 0 {
+		// Construct a new body with just the raw file and no callbacks. Since
+		// all progress tracking happens when the net.http code copies our
+		// request body into a new request, we can safely make this request
+		// outside of the flow of the transfer adapter, and if it fails, the
+		// transfer progress will be rewound at the top level
+		f, _ := os.OpenFile(t.Path, os.O_RDONLY, 0644)
```

Comment:
I think that we already have the file open by the time that we call this function. Can you call `io.Seeker.Seek(0, io.SeekStart) on the _file_ instead of on the `io.Reader` in the request?

Reply:
Unfortunately the issue in this case is that the underlying file has already been closed, so we can't issue a new request without re-opening it.